src/DataFramesMeta.jl

```diff
@@ -295,20 +299,46 @@ end
 ##
 ##############################################################################
 
-where(d::AbstractDataFrame, arg) = d[arg, :]
-where(d::AbstractDataFrame, f::Function) = d[f(d), :]
-where(g::GroupedDataFrame, f::Function) = g[Bool[f(x) for x in g]]
+function where_helper(x, args...)
+    t = (fun_to_vec(arg; nolhs = true, gensym_names = true) for arg in args)
+    quote
+        $where($x, $(t...))
+    end
+end
+
+and(x, y) = x .& y
+
+function df_to_bool(res)
+    if any(t -> !(t isa Union{AbstractVector{<:Union{Missing, Bool}}, BitArray{1}}), eachcol(res))
+        throw(ArgumentError("All arguments in @where must return a " *
+                            "AbstractVector{<:Union{Missing, Bool} or a BitArray{1}"))
```

Comment:
`BitArray{1} <: AbstractVector{<:Union{Missing, Bool}` already:
```suggestion
    if any(t -> !(t isa AbstractVector{<:Union{Missing, Bool}}), eachcol(res))
        throw(ArgumentError("All arguments in @where must return an " *
                            "AbstractVector{<:Union{Missing, Bool}}"))
```